src/types/decimal.rs

```diff
@@ -627,22 +648,36 @@ mod decimal_tests {
         //  Cf. https://randomascii.wordpress.com/2012/02/25/comparing-floating-point-numbers-2012-edition/
         let relative_error = ((diff_fractional - expected) / expected).abs();
         assert!(relative_error <= 10.0 * f64::EPSILON);
+
+        // diff 1.000_000_2 with 1.2
+        let (diff_integer, diff_fractional) = Decimal::difference_by_parts_lossy(
+            &Decimal::new(10_000_002, -7),
+            &Decimal::new(12, -1),
+        );
+        // 1 - 1 -> 0
+        assert_eq!(diff_integer, 0.into());
+        // 0.0000002 - 0.2 -> -0.1999998
+        let expected = -0.1999998;
+        // ideally, this would be based on ULPs, not epsilon
+        //  Cf. https://randomascii.wordpress.com/2012/02/25/comparing-floating-point-numbers-2012-edition/
+        let relative_error = ((diff_fractional - expected) / expected).abs();
+        assert!(relative_error <= 10.0 * f64::EPSILON);
```

Comment:
Learning opportunity for me: was `10.0` selected arbitrarily, or is that a best practice?